habitat/core/dataset.py

```diff
@@ -59,6 +59,18 @@ class Episode:
         default=None, validator=not_none_validator
     )
     info: Optional[Dict[str, str]] = None
+    _shortest_path_cache: Any = attr.ib(init=False, default=None)
```

Comment:
As I understand we are going to hide `shortest_path_cache` in Episode for ease of use. While it won't be visible during serialization.

Reply:
Yep.  Putting it on the episode seemed the cleanest, but we can't serialize the cache.